lib/langchain/vectorsearch/pgvector.rb

```diff
@@ -8,65 +8,74 @@ class Pgvector < Base
     # Gem requirements: gem "pgvector", "~> 0.2"
     #
     # Usage:
-    # pgvector = Langchain::Vectorsearch::Pgvector.new(url:, index_name:, llm:, llm_api_key:)
+    # pgvector = Langchain::Vectorsearch::Pgvector.new(url:, index_name:, llm:, namespace_column: nil, namespace: nil)
     #
 
     # The operators supported by the PostgreSQL vector search adapter
     OPERATORS = {
-      "cosine_distance" => "<=>",
-      "euclidean_distance" => "<->"
+      "cosine_distance" => "cosine",
+      "euclidean_distance" => "euclidean"
     }
     DEFAULT_OPERATOR = "cosine_distance"
 
-    attr_reader :operator, :quoted_table_name
+    attr_reader :db, :operator, :table_name, :namespace_column, :namespace, :documents_table
 
     # @param url [String] The URL of the PostgreSQL database
     # @param index_name [String] The name of the table to use for the index
     # @param llm [Object] The LLM client to use
-    # @param api_key [String] The API key for the Vectorsearch DB (not used for PostgreSQL)
-    def initialize(url:, index_name:, llm:, api_key: nil)
-      require "pg"
+    # @param namespace_column [String] The name of the column to use for the namespace
+    # @param namespace [String] The namespace to use for the index when inserting/querying
+    def initialize(url:, index_name:, llm:, namespace_column: nil, namespace: nil)
```

Comment:
@bborn Btw -- what are `namespace_column` and `namespace` used for in general?

Reply:
My use case: in AgentHQ, I let people create indices from various loader sources (files, rss, intercom, etc.). I want to store all the vectors in the same table (rather than a table per index), so I need a way (namespace) of differentiating them. 

Pinecone does similar: https://docs.pinecone.io/docs/namespaces

Reply:
Ah good point - yeah it's not necessary to make it configurable. Happy to make it hardcoded if you think that's better.

Reply:
Yeah, I think so!